src/main/java/nextstep/subway/line/domain/Sections.java

```diff
@@ -0,0 +1,149 @@
+            findSectionByUpStation(section.getUpStation())
+                    .ifPresent(it -> it.updateUpStation(section.getDownStation(), section.getDistance()));
+            sections.add(section);
+            return;
+        }
+
+        if (isDownStationExisted) {
+            findSectionByDownStation(section.getDownStation())
+                    .ifPresent(it -> it.updateDownStation(section.getUpStation(), section.getDistance()));
+            sections.add(section);
+            return;
+        }
+
+        throw new RuntimeException();
+    }
+
+    private void validateAlreadyExists(boolean isUpStationExisted, boolean isDownStationExisted) {
+        if (isUpStationExisted && isDownStationExisted) {
+            throw new IllegalArgumentException("이미 등록된 구간 입니다.");
+        }
+    }
+
+    private void validateNotExists(List<Station> stations, Section section) {
+        if (!stations.isEmpty()
+                && stations.stream().noneMatch(it -> it.equals(section.getDownStation()))
+                && stations.stream().noneMatch(it -> it.equals(section.getUpStation()))) {
+            throw new IllegalArgumentException("등록할 수 없는 구간 입니다.");
+        }
+    }
+
+    public List<Station> getStations() {
+        if (sections.isEmpty()) {
+            return Arrays.asList();
+        }
+
+        return getStationsInOrder();
+    }
+
+    private boolean isUpStationExisted(List<Station> stations, Station station) {
+        return stations.stream().anyMatch(it -> it.equals(station));
+    }
+
+    private boolean isDownStationExisted(List<Station> stations, Station station) {
+        return stations.stream().anyMatch(it -> it.equals(station));
+    }
+
+    private List<Station> getStationsInOrder() {
+        List<Station> stations = new ArrayList<>();
+        Station station = findUpFinalStation();
+
+        while (station != null) {
+            stations.add(station);
+            station = findNextStation(station);
+        }
+        return stations;
+    }
+
+    private Station findUpFinalStation() {
+        List<Station> upStations = sections.stream()
+                .map(Section::getUpStation).collect(Collectors.toList());
+        List<Station> downStations = sections.stream()
+                .map(Section::getDownStation).collect(Collectors.toList());
+        upStations.removeAll(downStations);
+
+        if (upStations.size() != 1) {
+            throw new NoSuchElementException("상행종점역을 찾을 수 없습니다.");
+        }
+        return upStations.get(0);
+    }
+
+    private Station findNextStation(Station station) {
+        return sections.stream()
+                .filter(section -> section.getUpStation().equals(station))
+                .findFirst()
+                .orElse(new Section())
+                .getDownStation();
+    }
+
+    private Optional<Section> findSectionByUpStation(Station station) {
+        return sections.stream()
+                .filter(it -> it.getUpStation().equals(station))
```

Comment:
```suggestion
                .filter(it -> it.matchUpStation(station))
```

전체적으로 getter를 사용해서 처리하는 부분에서 getter를 없애고 메시지를 전달해보면 어떨까요? 😄

Reply:
각 도메인으로 메시지를 던지는 방식으로 수정해보겠습니다~!